### PR TITLE
Remove Continuous vars with 1 value from histogram, add to info drawer

### PR DESCRIPTION
**Pull request by MDunitz**
## Need
- If an observation is an int type but it only has one value it shouldnt appear in a histogram

## Approach
- check if min/max match, if they match display in the infodrawer instead

Note that timepoint now appears in the info drawer rather than as a histogram

Screen Shot 2020-10-12 at 18 41 43
Screen Shot 2020-10-12 at 18 41 25

### CODE REVIEW

File: client/src/components/infoDrawer/infoDrawer.js [hunk 1]
```diff
@@ -49,14 +80,46 @@ class InfoDrawer extends PureComponent {
...
-            datasetTitle,
-            aboutURL,
-            singleValueCategories,
-            dataPortalProps,
-          }}
-        />
+        <Async
+          watchFn={InfoDrawer.watchAsync}
+          promiseFn={this.fetchAsyncProps}
+          watchProps={{ schema }}
```
Comment (seve) [inline client/src/components/infoDrawer/infoDrawer.js:86]: Will the schema ever change?  There's a minor edge case where a user clips to a single continuous value, how will the drawer react?

File: client/src/components/infoDrawer/infoDrawer.js [hunk 2]
```diff
@@ -49,14 +80,46 @@ class InfoDrawer extends PureComponent {
...
+        >
+          <Async.Pending>
+            <InfoFormat
+              {...{
+                datasetTitle,
+                aboutURL,
+                singleValueCategories,
+                dataPortalProps,
+              }}
+            />
```
Comment (seve) [inline client/src/components/infoDrawer/infoDrawer.js:96]: We need to reintroduce the loading state for the InfoFormat component (the skeleton + placeholder generation)

File: client/src/components/brushableHistogram/index.js [hunk 3]
```diff
@@ -650,7 +650,8 @@ class HistogramBrush extends React.PureComponent {
       Number.isNaN(summary.min) ||
       Number.isNaN(summary.max);
 
-    const OK2Render = !summary.categorical && !nonFiniteExtent;
+    const OK2Render =
+      !summary.categorical && !nonFiniteExtent && !isSingleValue;
```
Comment (seve) [inline client/src/components/brushableHistogram/index.js:654]: Can we make the `isSingleValue` check as early as possible and instead of assigning the value to `OK2Render` just do an early exit with the return value of `{OK2Render: false}` to avoid unused computes

**Review by seve — CHANGES_REQUESTED** [review]
Few questions about performance and edge cases

Annotation smoke tests are timing out for some reason, sometimes this means that the page is crashed or that something isn't where its expected and a querySelector fails to find its target.

**Comment by seve** [comment]
I just caused a little mischief, if you clip far down enough using the subset button and brushing the RSB histograms you can get all the continuous data to disappear and show in the drawer

**Comment by MDunitz** [comment]
Summary of slack discussion 
- We dont want to update the info drawer past the first render to avoid issues with continuous vars selected down to one value appearing in it
- We do need to make sure that the info drawer is populated by non selected (original) annoMatrix
- We dont want to rerun summarize or reload data
Solution
-  create a reducer, make the checks in the RSB, add the removed histograms to the redux state, and then in the info drawer just pull the values from the redux state 
- this puts all the checks in one place and allows us to (re)remove the async code from the info drawer

File: client/src/components/brushableHistogram/index.js [hunk 4]
```diff
@@ -620,7 +621,10 @@ class HistogramBrush extends React.PureComponent {
     const summary = column.summarize();
     const range = [summary.min, summary.max];
 
-    if (summary.min === summary.max && !isClipped) {
+    if (
+      (summary.min === summary.max && !isClipped) ||
+      singleContinuousValues.has(field)
```
Comment (seve) [inline client/src/components/brushableHistogram/index.js:626]: If the field is already in the singleContiousValues, we probably don't want to call dispatch again since the key/value are already in the map.  Might want to instead add an else if that just returns the same nulled object to avoid the unnecessary dispatch.

**Review by seve — CHANGES_REQUESTED** [review]
Nice fix for the edge case, one small piece of feedback

**Review by seve — APPROVED** [review]
LGTM!
